### ISSUE TITLE
Exception in UIRootUpdate relating to ideology building style

### ISSUE THREAD

**Issue by sealvessel** [cseelhoff/RimThreaded#709]
### **Describe the bug**
It seems if a ship turret is on fire, it produces this error.
Also seems to happen if a heat sink of standing lamp is on fire. 

It also seems that saving and reloading the save fixes the error but if a building catches on fire again, the error will repeat for all buildings on fire in space. 
### **Steps to reproduce the behavior (VERY IMPORTANT)**
Load the save. See that the error disappears as you use Explosion(Extinguisher) on everything on fire. 

### **Error Logs**
```c
Exception in UIRootUpdate: System.NullReferenceException: Object reference not set to an instance of an object
  at RimWorld.IdeoBuildingPresenceDemand.<BuildingPresent>b__17_0 (Verse.Thing t) [0x0000b] in <720a4e30fed64864bd8af057c96d4eb5>:0 
  at Verse.ListerBuildings.ColonistsHaveBuilding (System.Func`2[T,TResult] predicate) [0x00011] in <720a4e30fed64864bd8af057c96d4eb5>:0 
  at (wrapper dynamic-method) RimWorld.IdeoBuildingPresenceDemand.RimWorld.IdeoBuildingPresenceDemand.BuildingPresent_Patch1(RimWorld.IdeoBuildingPresenceDemand,Verse.Map)
  at RimWorld.Precept_Building+<GetAlerts>d__13.MoveNext () [0x00042] in <720a4e30fed64864bd8af057c96d4eb5>:0 
  at (wrapper dynamic-method) RimWorld.AlertsReadout.RimWorld.AlertsReadout.AlertsReadoutUpdate_Patch1(RimWorld.AlertsReadout)
  at (wrapper dynamic-method) RimWorld.UIRoot_Play.RimWorld.UIRoot_Play.UIRootUpdate_Patch1(RimWorld.UIRoot_Play)
UnityEngine.StackTraceUtility:ExtractStackTrace ()
(wrapper dynamic-method) Verse.Log:Verse.Log.Error_Patch2 (string)
(wrapper dynamic-method) RimWorld.UIRoot_Play:RimWorld.UIRoot_Play.UIRootUpdate_Patch1 (RimWorld.UIRoot_Play)
(wrapper dynamic-method) Verse.Root:Verse.Root.Update_Patch1 (Verse.Root)
Verse.Root_Play:Update ()
```

#### **Player.log**
!!! Link to contents of Player.log or Error.log (MUST BE LINK) !!!  
#### **HugsLib log**
https://gist.github.com/3fe25a4c3dffc18e1495703f42b3183d

### **Save file**
[Quicksave.rws.zip](https://github.com/cseelhoff/RimThreaded/files/7486590/Quicksave.rws.zip)

### **Screenshots**
<img width="2225" alt="image" src="https://user-images.githubusercontent.com/66351949/140524491-b88e6f5f-8854-4c3f-9e27-1a55e35ed245.png">
Standing lamp on fire
<img width="2443" alt="image" src="https://user-images.githubusercontent.com/66351949/140524997-8b9d63cc-d67f-4011-bdab-12ed44d9b4bc.png">
Ship turret on fire with errors. 

### **Mod list (Preferably a RimPy compatible list.)**
 <li>brrainz.harmony</li>
    <li>ludeon.rimworld</li>
    <li>ludeon.rimworld.royalty</li>
    <li>ludeon.rimworld.ideology</li>
    <li>unlimitedhugs.hugslib</li>
    <li>automatic.startupimpact</li>
    <li>telefonmast.graphicssettings</li>
    <li>edb.preparecarefully</li>
    <li>void.charactereditor</li>
    <li>oskarpotocki.vanillafactionsexpanded.core</li>
    <li>haplo.miscellaneous.core</li>
    <li>haplo.miscellaneous.robots</li>
    <li>alaestor.miscrobots.plusplus</li>
    <li>oskarpotocki.vfe.mechanoid</li>
    <li>roolo.runandgun</li>
    <li>roolo.searchanddestroy</li>
    <li>zylle.mapdesigner</li>
    <li>ykara.epoe</li>
    <li>taranchuk.archotechplus</li>
    <li>allencaine.tdpack</li>
    <li>gondragon.uinotincluded</li>
    <li>owlchemist.toggleableoverlays</li>
    <li>lwm.deepstorage</li>
    <li>legodude17.smartdecon</li>
    <li>vanillaexpanded.vcooke</li>
    <li>seohyeon.optimizationmeats</li>
    <li>owlchemist.barelyvisiblespots</li>
    <li>owlchemist.bettervanillamasking</li>
    <li>owlchemist.moonlight</li>
    <li>owlchemist.wallutilities</li>
    <li>owlchemist.perspectivebuildings</li>
    <li>owlchemist.perspectivetrees</li>
    <li>vanillaexpanded.vbrewecandt</li>
    <li>vanillaexpanded.vfeproduction</li>
    <li>oskarpotocki.vanillaexpanded.royaltypatches</li>
    <li>vanillaexpanded.vfecore</li>
    <li>vanillaexpanded.vfepower</li>
    <li>owlchemist.simplefx.smoke2</li>
    <li>owlchemist.ceilingutilities</li>
    <li>owlchemist.smartfarming</li>
    <li>owlchemist.windows</li>
    <li>fluffy.animaltab</li>
    <li>fluffy.areaunlocker</li>
    <li>fluffy.backuppower</li>
    <li>fluffy.blueprints</li>
    <li>fluffy.medicaltab</li>
    <li>fluffy.musicmanager</li>
    <li>fluffy.pharmacist</li>
    <li>fluffy.worktab</li>
    <li>dubwise.dubsappareltweaks</li>
    <li>dubwise.dubsmintmenus</li>
    <li>dubwise.dubsmintminimap</li>
    <li>dubwise.dubspaintshop</li>
    <li>arandomkiwi.rimsaves</li>
    <li>telardo.dragselect</li>
    <li>mehni.misc.modifications</li>
    <li>unlimitedhugs.allowtool</li>
    <li>madarauchiha.custommapsizes</li>
    <li>voult.betterpawncontrol</li>
    <li>automatic.bionicicons</li>
    <li>chippedchap.blueprinttotalstooltip</li>
    <li>brrainz.cameraplus</li>
    <li>dubwise.dubsbadhygiene</li>
    <li>sumghai.replimat</li>
    <li>sumghai.medpod</li>
    <li>avilmask.commonsense</li>
    <li>steaplay.constructvanillarockwalls</li>
    <li>unlimitedhugs.defensivepositions</li>
    <li>merthsoft.designatorshapes</li>
    <li>vanillaexpanded.vfefarming</li>
    <li>vanillaexpanded.vfesecurity</li>
    <li>vanillaexpanded.vfespacer</li>
    <li>owlchemist.cleanpathfinding</li>
    <li>owlchemist.permeableterrain</li>
    <li>frozensnowfox.improvedvanometricpowercells</li>
    <li>frozensnowfox.nodefaultshelfstorage</li>
    <li>frozensnowfox.growableambrosia</li>
    <li>frozensnowfox.filthvanisheswithrainandtime</li>
    <li>frozensnowfox.betterancientcomplexloot</li>
    <li>frozensnowfox.bettercamploot</li>
    <li>frozensnowfox.growablegrass</li>
    <li>frozensnowfox.efficientutilities</li>
    <li>mlie.giveupyourbuilding</li>
    <li>mlie.nomoreautoroof</li>
    <li>mlie.weaponssorter</li>
    <li>mlie.scenarioamender</li>
    <li>sarg.smartspeed</li>
    <li>neptimus7.homearearadius</li>
    <li>falconne.heatmap</li>
    <li>notfood.mendandrecycle</li>
    <li>modsyncrw.kv.rw</li>
    <li>com.github.alandariva.moreplanning</li>
    <li>ogre.ogrestack</li>
    <li>jaxe.reorderer</li>
    <li>cp.rimdeed</li>
    <li>cabbage.rimcities</li>
    <li>jaxe.rimhud</li>
    <li>sandy.rpgstyleinventory.avilmask.revamped</li>
    <li>denev.smartturretcovering</li>
    <li>ratys.rtfuse</li>
    <li>ratys.rtpowerswitch</li>
    <li>ratys.rtsolarflareshield</li>
    <li>neceros.fuseplus</li>
    <li>dubwise.dubsrimkit</li>
    <li>neitsa.preparelanding</li>
    <li>neptimus7.progressrenderer</li>
    <li>mastertea.randomplus</li>
    <li>herigony.mapedittools</li>
    <li>uuugggg.sharetheload</li>
    <li>dubwise.rimatomics</li>
    <li>dubwise.rimefeller</li>
    <li>dubwise.dubsskylights</li>
    <li>vanillaexpanded.vbookse</li>
    <li>vanillaexpanded.vwe</li>
    <li>vanillaexpanded.vweg</li>
    <li>vanillaexpanded.vwehw</li>
    <li>vanillaexpanded.vwel</li>
    <li>vanillaexpanded.vwec</li>
    <li>vanillaexpanded.vwenl</li>
    <li>legodude17.heavyweaponsreloading</li>
    <li>vanillaexpanded.vfemedical</li>
    <li>vanillaexpanded.vfeart</li>
    <li>vanillaexpanded.vfepropsanddecor</li>
    <li>vanillaexpanded.vfearchitect</li>
    <li>vanillaexpanded.vaeaccessories</li>
    <li>turnovus.submod.backpacksarenotbelts</li>
    <li>owlchemist.fridgeutilities</li>
    <li>vanillaexpanded.vbrewe</li>
    <li>vanillaexpanded.varme</li>
    <li>vanillaexpanded.vcef</li>
    <li>vanillaexpanded.vplantse</li>
    <li>spacemoth.vfavccrsze</li>
    <li>zzz.wallvitalsmonitor</li>
    <li>oskarpotocki.vanillaexpanded.ideologypatches</li>
    <li>anon.vanillaembrasures</li>
    <li>mlie.mechanicaldefense2walls</li>
    <li>chv.diagonalwalls</li>
    <li>jelly.rimflix</li>
    <li>murmur.walllight</li>
    <li>owlchemist.undergroundpowerconduits</li>
    <li>kentington.saveourship2</li>
    <li>kentington.saveourshipck</li>
    <li>dubwise.dubsperformanceanalyzer</li>
    <li>automatic.autocleaner</li>
    <li>oskarpotocki.vfe.insectoid</li>
    <li>bs.mbifvte</li>
    <li>imranfish.xmlextensions</li>
    <li>imranfishnikos.tallertabmenus</li>
    <li>orion.hospitality</li>
    <li>jkluch.haultostack</li>
    <li>mehni.pickupandhaul</li>
    <li>niilo007.niilosqol</li>
    <li>adambucior.betterelectronics</li>
    <li>kyd.rapidfireturrets</li>
    <li>trinity.runtimegcupdated</li>
    <li>majorhoff.rimthreaded</li>

**Comment by Niilo007** [comment]
please test if this is fixed in 2.5.16

**Comment by Niilo007** [comment]
im closing this because it has not been re tested and we will assume its fixed in 2.5.16.1
re open this issue if it still happens
